build_runner/lib/src/build_script_generate/build_script_generate.dart

```diff
@@ -10,35 +10,131 @@ import 'package:build_runner/build_runner.dart';
 import 'package:code_builder/code_builder.dart';
 import 'package:dart_style/dart_style.dart';
 
+import 'types.dart' as types;
+
 const scriptLocation = '.dart_tool/build/build.dart';
 
 Future<Null> ensureBuildScript() async {
   var scriptFile = new io.File(scriptLocation);
   // TODO - how can we invalidate this?
```

Comment:
do we have a bug for this?

Reply:
no, it's track in an internal design doc but not on github.

should we have a separate milestone for the script generation or keep putting them in the M0 milestone? We'll probably start collecting quite a few.

Reply:
I think we should keep them under M0, since it does fall under that

Reply:
https://github.com/dart-lang/build/issues/585